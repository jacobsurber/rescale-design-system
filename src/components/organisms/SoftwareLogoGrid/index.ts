export { SoftwareLogoGrid } from './SoftwareLogoGrid';
export type { SoftwareLogoGridProps, SoftwareItem } from './SoftwareLogoGrid';