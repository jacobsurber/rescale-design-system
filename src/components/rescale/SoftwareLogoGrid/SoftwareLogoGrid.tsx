import React, { useState } from 'react';
import { Avatar, Tooltip, Button, Popover } from 'antd';
import { PlusOutlined } from '@ant-design/icons';
import styled from 'styled-components';

export interface SoftwareItem {
  /** Unique identifier */
  id: string;
  /** Software name */
  name: string;
  /** Logo URL or React node */
  logo?: string | React.ReactNode;
  /** Software version */
  version?: string;
  /** Software description */
  description?: string;
  /** Category (e.g., 'CAD', 'CFD', 'FEA') */
  category?: string;
  /** Whether this software is featured */
  featured?: boolean;
}

export interface SoftwareLogoGridProps {
  /** Array of software items to display */
  items: SoftwareItem[];
  /** Maximum number of logos to show before "+X More" */
  maxVisible?: number;
  /** Size of the logos */
  size?: 'small' | 'default' | 'large';
  /** Whether to show software names below logos */
  showNames?: boolean;
  /** Whether logos should be clickable */
  clickable?: boolean;
  /** Click handler for individual software items */
  onItemClick?: (item: SoftwareItem) => void;
  /** Click handler for "+X More" button */
  onShowMore?: () => void;
  /** Custom className */
  className?: string;
  /** Custom styles */
  style?: React.CSSProperties;
}

const GridContainer = styled.div<{ $size: string }>`
  display: flex;
  flex-wrap: wrap;
  gap: ${props => {
    switch (props.$size) {
      case 'small': return 'var(--rescale-space-2)';
      case 'large': return 'var(--rescale-space-4)';
      default: return 'var(--rescale-space-3)';
    }
  }};
  align-items: center;
`;

const SoftwareItem = styled.div<{ 
  $size: string; 
  $clickable: boolean; 
  $showNames: boolean;
}>`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: var(--rescale-space-1);
  cursor: ${props => props.$clickable ? 'pointer' : 'default'};
  padding: ${props => {
    if (props.$showNames) {
      switch (props.$size) {
        case 'small': return 'var(--rescale-space-1)';
        case 'large': return 'var(--rescale-space-3)';
        default: return 'var(--rescale-space-2)';
      }
    }
    return '0';
  }};
  border-radius: var(--rescale-radius-base);
  transition: all var(--rescale-duration-fast) var(--rescale-easing-ease-in-out);
  
  &:hover {
    ${props => props.$clickable && `
      background: var(--rescale-color-gray-100);
      transform: translateY(-1px);
    `}
  }
`;

const SoftwareLogo = styled(Avatar)<{ $size: string }>`
  flex-shrink: 0;
  border: 1px solid var(--rescale-color-gray-300);
  
  ${props => {
    switch (props.$size) {
      case 'small':
        return `
          width: 32px !important;
          height: 32px !important;
          font-size: 14px;
        `;
      case 'large':
        return `
          width: 56px !important;
          height: 56px !important;
          font-size: 20px;
        `;
      default:
        return `
          width: 40px !important;
          height: 40px !important;
          font-size: 16px;
        `;
    }
  }}
  
  .ant-avatar-string {
    font-weight: var(--rescale-font-weight-semibold);
  }
`;

const SoftwareName = styled.span<{ $size: string }>`
  font-size: ${props => {
    switch (props.$size) {
      case 'small': return '10px';
      case 'large': return 'var(--rescale-font-size-sm)';
      default: return 'var(--rescale-font-size-xs)';
    }
  }};
  color: var(--rescale-color-gray-700);
  text-align: center;
  max-width: ${props => {
    switch (props.$size) {
      case 'small': return '40px';
      case 'large': return '72px';
      default: return '56px';
    }
  }};
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-weight: var(--rescale-font-weight-medium);
`;

const MoreButton = styled(Button)<{ $size: string }>`
  ${props => {
    switch (props.$size) {
      case 'small':
        return `
          width: 32px;
          height: 32px;
          font-size: 12px;
        `;
      case 'large':
        return `
          width: 56px;
          height: 56px;
          font-size: 16px;
        `;
      default:
        return `
          width: 40px;
          height: 40px;
          font-size: 14px;
        `;
    }
  }}
  
  border-radius: 50%;
  border: 2px dashed var(--rescale-color-gray-400);
  background: var(--rescale-color-gray-50);
  color: var(--rescale-color-gray-600);
  display: flex;
  align-items: center;
  justify-content: center;
  
  &:hover,
  &:focus {
    border-color: var(--rescale-color-brand-blue);
    color: var(--rescale-color-brand-blue);
    background: var(--rescale-color-light-blue);
  }
`;

const MoreCount = styled.span<{ $size: string }>`
  font-size: ${props => {
    switch (props.$size) {
      case 'small': return '10px';
      case 'large': return 'var(--rescale-font-size-xs)';
      default: return '11px';
    }
  }};
  font-weight: var(--rescale-font-weight-semibold);
  margin-top: 2px;
`;

const PopoverContent = styled.div`
  max-width: 280px;
`;

const PopoverGrid = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: var(--rescale-space-3);
  margin-bottom: var(--rescale-space-3);
`;

const PopoverItem = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: var(--rescale-space-1);
  padding: var(--rescale-space-2);
  border-radius: var(--rescale-radius-base);
  cursor: pointer;
  transition: background var(--rescale-duration-fast);
  
  &:hover {
    background: var(--rescale-color-gray-100);
  }
`;

const PopoverItemName = styled.span`
  font-size: var(--rescale-font-size-xs);
  color: var(--rescale-color-gray-700);
  text-align: center;
  max-width: 60px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

export const SoftwareLogoGrid: React.FC<SoftwareLogoGridProps> = ({
  items,
  maxVisible = 6,
  size = 'default',
  showNames = false,
  clickable = true,
  onItemClick,
  onShowMore,
  className,
  style,
}) => {
  const [popoverVisible, setPopoverVisible] = useState(false);
  
  const visibleItems = items.slice(0, maxVisible);
  const remainingItems = items.slice(maxVisible);
  const hasMore = remainingItems.length > 0;

  const handleItemClick = (item: SoftwareItem) => {
    if (clickable && onItemClick) {
      onItemClick(item);
    }
  };

  const handleMoreClick = () => {
    if (onShowMore) {
      onShowMore();
    } else {
      setPopoverVisible(!popoverVisible);
    }
  };

  const renderSoftwareItem = (item: SoftwareItem, index: number) => {
    const logoContent = typeof item.logo === 'string' ? (
      <SoftwareLogo
        $size={size}
        src={item.logo}
        alt={item.name}
      />
    ) : item.logo ? (
      <SoftwareLogo $size={size}>
        {item.logo}
      </SoftwareLogo>
    ) : (
      <SoftwareLogo $size={size}>
        {item.name.charAt(0).toUpperCase()}
      </SoftwareLogo>
    );

    const content = (
      <SoftwareItem
        key={item.id}
        $size={size}
        $clickable={clickable}
        $showNames={showNames}
        onClick={() => handleItemClick(item)}
      >
        {logoContent}
        {showNames && (
          <SoftwareName $size={size}>
            {item.name}
          </SoftwareName>
        )}
      </SoftwareItem>
    );

    if (clickable && (item.description || item.version)) {
      const tooltipTitle = (
        <div>
          <div style={{ fontWeight: 600 }}>{item.name}</div>
          {item.version && (
            <div style={{ fontSize: '12px', color: '#666' }}>
              Version {item.version}
            </div>
          )}
          {item.description && (
            <div style={{ fontSize: '12px', marginTop: '4px' }}>
              {item.description}
            </div>
          )}
          {item.category && (
            <div style={{ fontSize: '11px', color: '#888', marginTop: '4px' }}>
              Category: {item.category}
            </div>
          )}
        </div>
      );

      return (
        <Tooltip key={item.id} title={tooltipTitle} placement="top">
          {content}
        </Tooltip>
      );
    }

    return content;
  };

  const morePopoverContent = (
    <PopoverContent>
      <PopoverGrid>
        {remainingItems.map((item) => (
          <PopoverItem
            key={item.id}
            onClick={() => {
              handleItemClick(item);
              setPopoverVisible(false);
            }}
          >
            {typeof item.logo === 'string' ? (
              <Avatar size={32} src={item.logo} alt={item.name} />
            ) : item.logo ? (
              <Avatar size={32}>{item.logo}</Avatar>
            ) : (
              <Avatar size={32}>{item.name.charAt(0).toUpperCase()}</Avatar>
            )}
            <PopoverItemName>{item.name}</PopoverItemName>
          </PopoverItem>
        ))}
      </PopoverGrid>
    </PopoverContent>
  );

  return (
    <GridContainer $size={size} className={className} style={style}>
      {visibleItems.map((item, index) => renderSoftwareItem(item, index))}
      
      {hasMore && (
        <div>
          {onShowMore ? (
            <MoreButton
              $size={size}
              type="text"
              onClick={handleMoreClick}
            >
              <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
                <PlusOutlined />
                <MoreCount $size={size}>{remainingItems.length}</MoreCount>
              </div>
            </MoreButton>
          ) : (
            <Popover
              content={morePopoverContent}
              title={`+${remainingItems.length} More Software`}
              trigger="click"
              open={popoverVisible}
              onOpenChange={setPopoverVisible}
              placement="bottomLeft"
            >
              <MoreButton
                $size={size}
                type="text"
                onClick={handleMoreClick}
              >
                <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
                  <PlusOutlined />
                  <MoreCount $size={size}>{remainingItems.length}</MoreCount>
                </div>
              </MoreButton>
            </Popover>
          )}
        </div>
      )}
    </GridContainer>
  );
};

export default SoftwareLogoGrid;