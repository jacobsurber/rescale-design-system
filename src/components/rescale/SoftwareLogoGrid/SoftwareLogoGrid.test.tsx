import { screen, fireEvent } from '@testing-library/react';
import { SoftwareLogoGrid } from './SoftwareLogoGrid';
import { 
  render, 
  commonTestScenarios, 
  runAxeTest,
  mockData,
  snapshotHelpers,
} from '../../../test-utils';

describe('SoftwareLogoGrid', () => {
  const mockSoftware = mockData.generateSoftwareItems(8);
  const defaultProps = {
    items: mockSoftware,
    maxVisible: 6,
  };

  // Common test scenarios
  commonTestScenarios.shouldRender(SoftwareLogoGrid, defaultProps);
  commonTestScenarios.shouldBeAccessible(SoftwareLogoGrid, defaultProps);
  commonTestScenarios.shouldWorkWithThemes(SoftwareLogoGrid, defaultProps);
  commonTestScenarios.shouldBeResponsive(SoftwareLogoGrid, defaultProps);

  describe('Item Display', () => {
    it('displays all items when count is below maxVisible', () => {
      const items = mockData.generateSoftwareItems(3);
      render(<SoftwareLogoGrid items={items} maxVisible={6} />);
      
      items.forEach(item => {
        expect(screen.getByText(item.name.charAt(0).toUpperCase())).toBeInTheDocument();
      });
    });

    it('displays maxVisible items plus "+X More" button when items exceed limit', () => {
      render(<SoftwareLogoGrid items={mockSoftware} maxVisible={4} />);
      
      // Should show 4 visible items
      const visibleItems = screen.getAllByRole('button');
      expect(visibleItems).toHaveLength(5); // 4 items + 1 "+X More" button
      
      // Should show "+4 More" button
      expect(screen.getByText('4')).toBeInTheDocument();
    });

    it('shows software names when showNames is true', () => {
      const items = mockData.generateSoftwareItems(3);
      render(<SoftwareLogoGrid items={items} showNames />);
      
      items.forEach(item => {
        expect(screen.getByText(item.name)).toBeInTheDocument();
      });
    });

    it('hides software names when showNames is false', () => {
      const items = mockData.generateSoftwareItems(3);
      render(<SoftwareLogoGrid items={items} showNames={false} />);
      
      items.forEach(item => {
        expect(screen.queryByText(item.name)).not.toBeInTheDocument();
      });
    });
  });

  describe('Size Variants', () => {
    it('renders small size correctly', () => {
      const { container } = render(
        <SoftwareLogoGrid items={mockSoftware} size="small" />
      );
      
      const logoElements = container.querySelectorAll('.ant-avatar');
      logoElements.forEach(logo => {
        expect(logo).toHaveStyle({ width: '32px', height: '32px' });
      });
    });

    it('renders default size correctly', () => {
      const { container } = render(
        <SoftwareLogoGrid items={mockSoftware} size="default" />
      );
      
      const logoElements = container.querySelectorAll('.ant-avatar');
      logoElements.forEach(logo => {
        expect(logo).toHaveStyle({ width: '40px', height: '40px' });
      });
    });

    it('renders large size correctly', () => {
      const { container } = render(
        <SoftwareLogoGrid items={mockSoftware} size="large" />
      );
      
      const logoElements = container.querySelectorAll('.ant-avatar');
      logoElements.forEach(logo => {
        expect(logo).toHaveStyle({ width: '56px', height: '56px' });
      });
    });
  });

  describe('Interaction', () => {
    it('calls onItemClick when item is clicked and clickable is true', async () => {
      const onItemClick = jest.fn();
      const items = mockData.generateSoftwareItems(3);
      
      const { user } = render(
        <SoftwareLogoGrid 
          items={items} 
          clickable 
          onItemClick={onItemClick} 
        />
      );
      
      const firstItem = screen.getAllByRole('button')[0];
      await user.click(firstItem);
      
      expect(onItemClick).toHaveBeenCalledWith(items[0]);
    });

    it('does not call onItemClick when clickable is false', async () => {
      const onItemClick = jest.fn();
      const items = mockData.generateSoftwareItems(3);
      
      const { user } = render(
        <SoftwareLogoGrid 
          items={items} 
          clickable={false} 
          onItemClick={onItemClick} 
        />
      );
      
      const itemElements = screen.getAllByRole('button');
      if (itemElements.length > 0) {
        await user.click(itemElements[0]);
        expect(onItemClick).not.toHaveBeenCalled();
      }
    });

    it('shows popover when "+X More" button is clicked', async () => {
      const { user } = render(
        <SoftwareLogoGrid items={mockSoftware} maxVisible={4} />
      );
      
      const moreButton = screen.getByRole('button', { name: /\+\d+ More/ });
      await user.click(moreButton);
      
      // Check if popover content appears
      expect(screen.getByText(/\+\d+ More Software/)).toBeInTheDocument();
    });

    it('calls onShowMore when provided instead of showing popover', async () => {
      const onShowMore = jest.fn();
      const { user } = render(
        <SoftwareLogoGrid 
          items={mockSoftware} 
          maxVisible={4} 
          onShowMore={onShowMore} 
        />
      );
      
      const moreButton = screen.getByRole('button', { name: /\+\d+ More/ });
      await user.click(moreButton);
      
      expect(onShowMore).toHaveBeenCalled();
    });
  });

  describe('Tooltips', () => {
    it('shows tooltip with software details on hover when clickable', async () => {
      const items = mockData.generateSoftwareItems(1);
      items[0].description = 'Test description';
      items[0].version = '1.0.0';
      
      const { user } = render(
        <SoftwareLogoGrid items={items} clickable />
      );
      
      const itemButton = screen.getByRole('button');
      await user.hover(itemButton);
      
      // Tooltip should appear with software details
      expect(screen.getByText(items[0].name)).toBeInTheDocument();
      expect(screen.getByText('Version 1.0.0')).toBeInTheDocument();
      expect(screen.getByText('Test description')).toBeInTheDocument();
    });

    it('does not show tooltip when item has no additional details', async () => {
      const items = [
        {
          id: 'test',
          name: 'Test Software',
          logo: 'ðŸ§ª',
        }
      ];
      
      const { user } = render(
        <SoftwareLogoGrid items={items} clickable />
      );
      
      const itemButton = screen.getByRole('button');
      await user.hover(itemButton);
      
      // Only the logo/name should be visible, no tooltip
      expect(screen.queryByText('Version')).not.toBeInTheDocument();
    });
  });

  describe('Logo Types', () => {
    it('renders string logos correctly', () => {
      const items = [
        {
          id: 'test',
          name: 'Test',
          logo: 'https://example.com/logo.png',
        }
      ];
      
      const { container } = render(<SoftwareLogoGrid items={items} />);
      
      const img = container.querySelector('img');
      expect(img).toHaveAttribute('src', 'https://example.com/logo.png');
    });

    it('renders React node logos correctly', () => {
      const items = [
        {
          id: 'test',
          name: 'Test',
          logo: <span data-testid="custom-logo">Custom</span>,
        }
      ];
      
      render(<SoftwareLogoGrid items={items} />);
      
      expect(screen.getByTestId('custom-logo')).toBeInTheDocument();
    });

    it('renders fallback when no logo provided', () => {
      const items = [
        {
          id: 'test',
          name: 'Test Software',
        }
      ];
      
      render(<SoftwareLogoGrid items={items} />);
      
      expect(screen.getByText('T')).toBeInTheDocument();
    });
  });

  describe('Popover Content', () => {
    it('displays all remaining items in popover grid', async () => {
      const items = mockData.generateSoftwareItems(8);
      const { user } = render(
        <SoftwareLogoGrid items={items} maxVisible={4} />
      );
      
      const moreButton = screen.getByRole('button', { name: /\+4/ });
      await user.click(moreButton);
      
      // Should show the 4 remaining items in popover
      const popoverItems = screen.getAllByTestId(/popover-item-/);
      expect(popoverItems).toHaveLength(4);
    });

    it('calls onItemClick when popover item is clicked', async () => {
      const onItemClick = jest.fn();
      const items = mockData.generateSoftwareItems(6);
      const { user } = render(
        <SoftwareLogoGrid 
          items={items} 
          maxVisible={4} 
          onItemClick={onItemClick}
          clickable
        />
      );
      
      const moreButton = screen.getByRole('button', { name: /\+2/ });
      await user.click(moreButton);
      
      const popoverItems = screen.getAllByTestId(/popover-item-/);
      await user.click(popoverItems[0]);
      
      expect(onItemClick).toHaveBeenCalledWith(items[4]); // First item in overflow
    });
  });

  describe('Empty State', () => {
    it('renders empty grid when no items provided', () => {
      const { container } = render(<SoftwareLogoGrid items={[]} />);
      
      expect(container.firstChild).toBeEmptyDOMElement();
    });

    it('does not show "+X More" button when no overflow items', () => {
      const items = mockData.generateSoftwareItems(3);
      render(<SoftwareLogoGrid items={items} maxVisible={6} />);
      
      expect(screen.queryByText(/\+\d+/)).not.toBeInTheDocument();
    });
  });

  describe('Accessibility', () => {
    it('has proper ARIA labels for interactive items', async () => {
      const items = mockData.generateSoftwareItems(3);
      const { container } = render(
        <SoftwareLogoGrid items={items} clickable />
      );
      
      await runAxeTest(container);
    });

    it('supports keyboard navigation', async () => {
      const onItemClick = jest.fn();
      const items = mockData.generateSoftwareItems(3);
      
      const { user } = render(
        <SoftwareLogoGrid 
          items={items} 
          clickable 
          onItemClick={onItemClick} 
        />
      );
      
      // Tab to first item
      await user.tab();
      expect(screen.getAllByRole('button')[0]).toHaveFocus();
      
      // Enter should trigger click
      await user.keyboard('{Enter}');
      expect(onItemClick).toHaveBeenCalledWith(items[0]);
    });

    it('has proper role and labels for "+X More" button', () => {
      render(<SoftwareLogoGrid items={mockSoftware} maxVisible={4} />);
      
      const moreButton = screen.getByRole('button', { name: /\+4/ });
      expect(moreButton).toBeInTheDocument();
      expect(moreButton).toHaveAttribute('aria-label', expect.stringContaining('Show more'));
    });
  });

  describe('Snapshots', () => {
    it('matches snapshot with few items', () => {
      const items = mockData.generateSoftwareItems(3);
      snapshotHelpers.createComponentSnapshot(SoftwareLogoGrid, {
        items,
        maxVisible: 6,
        showNames: true,
      });
    });

    it('matches snapshot with overflow', () => {
      snapshotHelpers.createComponentSnapshot(SoftwareLogoGrid, {
        items: mockSoftware,
        maxVisible: 4,
        showNames: false,
      });
    });

    it('matches snapshot in different sizes', () => {
      const items = mockData.generateSoftwareItems(4);
      
      ['small', 'default', 'large'].forEach(size => {
        snapshotHelpers.createComponentSnapshot(SoftwareLogoGrid, {
          items,
          size: size as 'small' | 'default' | 'large',
          showNames: true,
        });
      });
    });
  });
});