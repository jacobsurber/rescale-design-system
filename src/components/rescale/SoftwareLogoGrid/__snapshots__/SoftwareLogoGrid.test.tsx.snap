// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SoftwareLogoGrid Snapshots matches snapshot in different sizes 1`] = `
<div
  class="sc-bRKDuR llfmNz"
>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm GyTs"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv dlpNph css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="ANSYS Fluent"
        src="ðŸŒŠ"
      />
    </span>
    <span
      class="sc-ggWZvA crbPDy"
    >
      ANSYS Fluent
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm GyTs"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv dlpNph css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="OpenFOAM"
        src="ðŸ’¨"
      />
    </span>
    <span
      class="sc-ggWZvA crbPDy"
    >
      OpenFOAM
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm GyTs"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv dlpNph css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="Abaqus"
        src="ðŸ”§"
      />
    </span>
    <span
      class="sc-ggWZvA crbPDy"
    >
      Abaqus
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm GyTs"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv dlpNph css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="MATLAB"
        src="ðŸ“Š"
      />
    </span>
    <span
      class="sc-ggWZvA crbPDy"
    >
      MATLAB
    </span>
  </div>
</div>
`;

exports[`SoftwareLogoGrid Snapshots matches snapshot in different sizes 2`] = `
<div
  class="sc-bRKDuR cfUdnq"
>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm biOyrH"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="ANSYS Fluent"
        src="ðŸŒŠ"
      />
    </span>
    <span
      class="sc-ggWZvA hdWqAX"
    >
      ANSYS Fluent
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm biOyrH"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="OpenFOAM"
        src="ðŸ’¨"
      />
    </span>
    <span
      class="sc-ggWZvA hdWqAX"
    >
      OpenFOAM
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm biOyrH"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="Abaqus"
        src="ðŸ”§"
      />
    </span>
    <span
      class="sc-ggWZvA hdWqAX"
    >
      Abaqus
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm biOyrH"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="MATLAB"
        src="ðŸ“Š"
      />
    </span>
    <span
      class="sc-ggWZvA hdWqAX"
    >
      MATLAB
    </span>
  </div>
</div>
`;

exports[`SoftwareLogoGrid Snapshots matches snapshot in different sizes 3`] = `
<div
  class="sc-bRKDuR hTvHoB"
>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm iHdBZO"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv dvyqMm css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="ANSYS Fluent"
        src="ðŸŒŠ"
      />
    </span>
    <span
      class="sc-ggWZvA gZZFkY"
    >
      ANSYS Fluent
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm iHdBZO"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv dvyqMm css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="OpenFOAM"
        src="ðŸ’¨"
      />
    </span>
    <span
      class="sc-ggWZvA gZZFkY"
    >
      OpenFOAM
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm iHdBZO"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv dvyqMm css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="Abaqus"
        src="ðŸ”§"
      />
    </span>
    <span
      class="sc-ggWZvA gZZFkY"
    >
      Abaqus
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm iHdBZO"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv dvyqMm css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="MATLAB"
        src="ðŸ“Š"
      />
    </span>
    <span
      class="sc-ggWZvA gZZFkY"
    >
      MATLAB
    </span>
  </div>
</div>
`;

exports[`SoftwareLogoGrid Snapshots matches snapshot with few items 1`] = `
<div
  class="sc-bRKDuR cfUdnq"
>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm biOyrH"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="ANSYS Fluent"
        src="ðŸŒŠ"
      />
    </span>
    <span
      class="sc-ggWZvA hdWqAX"
    >
      ANSYS Fluent
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm biOyrH"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="OpenFOAM"
        src="ðŸ’¨"
      />
    </span>
    <span
      class="sc-ggWZvA hdWqAX"
    >
      OpenFOAM
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm biOyrH"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="Abaqus"
        src="ðŸ”§"
      />
    </span>
    <span
      class="sc-ggWZvA hdWqAX"
    >
      Abaqus
    </span>
  </div>
</div>
`;

exports[`SoftwareLogoGrid Snapshots matches snapshot with overflow 1`] = `
<div
  class="sc-bRKDuR cfUdnq"
>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm joHOqy"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="ANSYS Fluent"
        src="ðŸŒŠ"
      />
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm joHOqy"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="OpenFOAM"
        src="ðŸ’¨"
      />
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm joHOqy"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="Abaqus"
        src="ðŸ”§"
      />
    </span>
  </div>
  <div
    aria-describedby="test-id"
    class="sc-hvigdm joHOqy"
  >
    <span
      class="ant-avatar ant-avatar-circle ant-avatar-image sc-fhHczv fTkduH css-dev-only-do-not-override-5rmn7t"
    >
      <img
        alt="MATLAB"
        src="ðŸ“Š"
      />
    </span>
  </div>
  <div>
    <button
      aria-describedby="test-id"
      class="ant-btn css-dev-only-do-not-override-5rmn7t ant-btn-text ant-btn-color-default ant-btn-variant-text sc-dTvVRJ jeuxjj"
      type="button"
    >
      <div
        style="display: flex; flex-direction: column; align-items: center;"
      >
        <span
          aria-label="plus"
          class="anticon anticon-plus"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </span>
        <span
          class="sc-jwTyAe gARIeo"
        >
          4
        </span>
      </div>
    </button>
  </div>
</div>
`;
